src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -104,6 +104,90 @@ String getLogStreamName(String thingName) {
         return logStreamName.replace(":", "+");
     }
 
+    /**
+     * Gets CW input log events from the component which processSingleLogFile need to be uploaded to CloudWatch.
+     *
+     * @param attempt a cloudwatch attempt
+     * @param cpInfo log files information for a component to read logs from.
+     * @param fileInfo metadata of the file
+     * @param chan a byte channel that allows to read a file from the specified position
+     */
+    public CloudWatchAttempt processSingleLogFile(CloudWatchAttempt attempt, ComponentLogFileInformation cpInfo,
```

Comment:
Are you going to update the tests that call the old method to call this one on another pr?

Reply:
Separate PR